integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+    public void testCTE() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id).as(select(document.id).from(document))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEWithBinds() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, select(new Binds<IdHolderCTE>().bind(idHolderCTE.id, document.id)).from(document))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEWithBinds2() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                .with(idHolderCTE, select(
+                        bind(idHolderCTE.id, document.id)).from(document))
+                .select(idHolderCTE.id).from(idHolderCTE)
+                .fetch();
+
+                assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQL.class })
+    public void testCTEUnion() {
+        doInJPA(entityManager -> {
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id).as(union(select(document.id).from(document), intersect(select(document.id).from(document), select(document.id).from(document))))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class })
+    public void testCTEFromValues() {
+        doInJPA(entityManager -> {
+            Document theBook = new Document();
+            theBook.setId(1337L);
+            theBook.setName("test");
+
+            List<Long> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .with(idHolderCTE, idHolderCTE.id).as(new BlazeJPAQuery<Document>(entityManager, cbf)
+                            .fromValues(document, Collections.singleton(theBook))
+                            .select(document.id))
+                    .select(idHolderCTE.id).from(idHolderCTE)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoMySQLOld.class, NoDB2.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet` and also `// NOTE: Recursive CTE support was only added in MySQL8`.
Apart from that, why do you exclude DB2 here?

Reply:
Perhaps its not supported because I used join/on, instead of from/where. Including the test for DB2 again to see if it still fails with the change.

Reply:
I know DB2 has an issue with joins in recursive queries, but please add a note if that is the issue here.